core/domain/classifier_domain.py

```diff
@@ -277,6 +277,27 @@ def status(self):
     def training_data(self):
         return self._training_data
 
+    def update_state_name(self, state_name):
```

Comment:
@seanlip @prasanna08 . That is true. If we completely establish that its not mutable, I can remove it now. But, I don't think there is an issue in keeping it around and wait for the end-to-end framework to be implemented. We can remove it then if its not required.